core/src/main/scala/scala/pickling/generator/ast.scala

```diff
@@ -0,0 +1,202 @@
+package scala.pickling
+package generator
+
+
+
+// 1. Read symbols for type
+// 2. Use an algorithm to determine the behavior of the pickle/unpickle methods (i..e generate an AST)
+//     CallConstructor(V)
+//     SetField(<date>)
+//     SetField(<id>)
+//     ---------------
+//     GetField(<date>)
+//     GetField(<id>)
+// 3. Generate code using IrAst
+
+
+/**
+ * An AST representing simple pickling behavior.  We generate this before we generate the implementing code.
+ *
+ *
+ */
+sealed trait IrAst {
+  /** Returns true if the implementation of this particular AST node requires the use of of
+    * a private member, for which we do not have access.
+    * @return
+    */
+  def requiresReflection: Boolean
```

Comment:
If I read the list of all members during the compile-time that's still using Reflection API but that's safe. What's denoted here is the use of runtime reflection? I don't know if this also denotes the level of safety as well.